scipy/interpolate/interpolate.py

```diff
@@ -75,19 +84,26 @@ def lagrange(x, w):
     ...          label=r"$3 x^2 - 2 x$", linestyle='-.')
     >>> plt.legend()
     >>> plt.show()
-
     """
 
     M = len(x)
     p = poly1d(0.0)
     for j in range(M):
-        pt = poly1d(w[j])
+        pt = poly1d(y[j])
         for k in range(M):
             if k == j:
                 continue
             fac = x[j]-x[k]
             pt *= poly1d([1.0, -x[k]])/fac
         p += pt
+    diff = M - len(p.coeffs)
```

Comment:
This is only needed in the `if not reduce` branch, so best move this line over there